lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+# logging and default cache behavior
+
+- cloudfront_distribution:
+    state: present
+    caller_reference: unique test distribution id
+    origins:
+        - id: 'my test origin-000111'
+          domain_name: www.example.com
+          origin_path: /production
+          custom_headers:
+            - header_name: MyCustomHeaderName
+              header_value: MyCustomHeaderValue
+    default_cache_behavior:
+      target_origin_id: 'my test origin-000111'
+      forwarded_values:
+        query_string: true
+        cookies:
+          forward: all
+        headers:
+         - '*'
+      viewer_protocol_policy: allow-all
+      smooth_streaming: true
+      compress: true
+      allowed_methods:
+        items:
+          - GET
+          - HEAD
+        cached_methods:
+          - GET
+          - HEAD
+    logging:
+      enabled: true
+      include_cookies: false
+      bucket: mylogbucket.s3.amazonaws.com
+      prefix: myprefix/
+    enabled: false
+    comment: this is a cloudfront distribution with logging
+
+# delete a distribution
+
+- cloudfront_distribution:
+    state: absent
+    distribution_id: E1ZNUV0U7KWO4P
+
+# create a presigned url for a distribution based on a
+# distribution_id and from a local pem file
+
+- cloudfront_distribution:
+    generate_presigned_url: yes
+    distribution_id: E1RP5A2MJ8073O
+    pem_key_path: /home/user/ansible/pk-APKAJMTT6OPAZSFTNSCZ.pem
+    cloudfront_url_to_sign: 'http://d3vodljfucvmwf.cloudfront.net/example.txt'
+    presigned_url_expire_date: '2017-04-20'
+
+# create a streaming distribution
+
+- cloudfront_distribution:
+     streaming_distribution: yes
+     state: present
+     default_s3_origin_domain_name: my-example-bucket.s3.amazonaws.com
+     comment: example streaming distribution
+
+# create a streaming distribution with tags
+
+- cloudfront_distribution:
+     streaming_distribution: yes
+     default_s3_origin_domain_name: example-bucket.s3.amazonaws.com
+     caller_reference: test streaming
+     comment: example streaming distribution
+     tags:
+       - Name: example distribution
+       - Project: example project
+       - Priority: 1
+
+# update a streaming distribution
+
+- cloudfront_distribution:
+    streaming_distribution: yes
+    state: present
+    streaming_distribution_id: E2RTIUCAA9RINU
+    comment: modified streaming distribution
+
+'''
+
+RETURN = '''
+
+location:
+    description: describes a url specifying the output of the action just run.
+    returned: applies to C(streaming_distribution=no) with C(state=present)
+      or when specifying C(generate_presigned_url=yes)
+    type: str
+
+'''
+
+from ansible.module_utils.ec2 import get_aws_connection_info
+from ansible.module_utils.ec2 import ec2_argument_spec, boto3_conn, HAS_BOTO3
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible.module_utils.basic import AnsibleModule
+from ansible.modules.cloud.amazon.cloudfront_facts import (
+    CloudFrontFactsServiceManager)
```

Comment:
This isn't trivial to remove as there are 18 instances of cloudfront_facts methods being called and would result in duplicate code between the two modules. It looks like shippable is failing on importing this module this too now. If it's a show-stopper, I can implement the required functions but would prefer not to if I can avoid it.